### DISCUSSION TITLE
Doesn't work with java 8 (open jdk)

### DISCUSSION THREAD

**Discussion by algos320** [Majlanky/couch-slacker#89] [Q&A]
Hi all, when I tried to use library in the project with Java 8(open jdk)I saw BindExceptions "couchdb.client".
Maybe error due to jackarta package in class CouchDbProperties.

**Comment by Majlanky** [comment]
Hi @algos320,

can u provide more information on what "use library" mean, provide whole exception stack and pom.xml/gradle.build? The library is developed on Azul which is OpenJDK-based distribution so I need as much information as possible to find the origin of the problem. Thx

**Comment by algos320** [comment]
@Majlanky I mean under "use library" add dependency to spring boot project to connect to couch db, I did everything that needs from readme
[stack.txt](https://github.com/Majlanky/couch-slacker/files/11628084/stack.txt)
[pom.txt](https://github.com/Majlanky/couch-slacker/files/11628095/pom.txt)

**Comment by Majlanky** [comment]
Hi @algos320 
Now it is obvious. It is just simple exception about missing bean. The missing bean is CouchDbClient which is prepared as part of CouchSlackerConfiguration. What you probably missed is:
```java
@Configuration
class AppConfig extends CouchSlackerConfiguration {

}
```
In this matter is the library kinda oldfassioned and its missing autoconfiguration. The other possibility is to add factory method for the bean as declared in the config class:
```java
@Bean(destroyMethod = "close")
    public CouchDbClient dbClient(@NotNull CouchDbProperties properties,
                                  @Nullable @Autowired(required = false) List<IdGenerator<?>> idGenerators,
                                  @NotNull @Lazy SpringCouchDbContext dbContext,
                                  @Nullable @Autowired(required = false) ObjectMapper objectMapper) {
        Assert.notNull(properties, "Properties must not be null.");
        return CouchDbClient.builder().properties(properties).idGenerators(idGenerators).dbContext(dbContext).objectMapper(objectMapper).build();
    }
```

**Comment by Majlanky** [comment]
Btw if u like the lib, I really appreciate the star 😄

**Comment by algos320** [comment]
Hi @Majlanky ,  I tried to create the configuration class but I see this error
[stack.txt](https://github.com/Majlanky/couch-slacker/files/11630641/stack.txt)
My application yaml:
[yaml.txt](https://github.com/Majlanky/couch-slacker/files/11630671/yaml.txt)

**Comment by Majlanky** [comment]
@algos320 this looks like your internal problem. This bind exception is usually problem cased by the configuration. If u are interested, provide me whole project and I will investigate what is wrong because textually it looks ok.
